insights/parsers/samba_logs.py

```diff
@@ -0,0 +1,82 @@
+"""
+===================================================
+samba logs - files matching ``/var/log/samba/*.log``
+===================================================
+"""
+
+from insights import parser, LogFileOutput
+from datetime import datetime
+from insights.specs import Specs
+
+
+def strip_surrounds(s):
+    start = 0
+    end = len(s) - 1
+    while (s[start] == '[' and s[end] == ']') or \
+      (s[start] == '[' and s[end] == ',') or \
+      (s[start] == '(' and s[end] == ')'):
+        start += 1
+        end -= 1
+    if s[end] == ',':
+        end -= 1
+    return s[start:(end + 1)]
+
+
+@parser(Specs.samba_logs)
+class SAMBALog(LogFileOutput):
+    """
+    Parser class for reading samba log files.  The main work is done by the
+    LogFileOutput super-class.
+
+    .. note:
+
+        Please refer to its super-class :class:`insights.core.LogFileOutput`
+
+    Sample input:
+
+       [2018/12/07 07:09:44.812154,  5, pid=6434, effective(0, 0), real(0, 0)] ../source3/param/loadparm.c:1344(free_param_opts)
```

Comment:
One more point, seems the two lines in the `Sample` should be one line in the real world, would you please, remove the `<CR> (newline)` and put each of them in single lines?
Sorry for missing this in former review.

Reply:
@amitkumar50 - would you please change it per above comment?

Reply:
> @amitkumar50 - would you please change it per above comment?

@xiangce Actually in log.smbd these are 2 lines separated by <CR>, taken from real world log.
  Freeing parametrics:
[2020/01/14 18:37:33.499290,  3] ../../source3/param/loadparm.c:550(init_globals)
  Initialising global parameters
[2020/01/14 18:37:33.499368,  3] ../../source3/param/loadparm.c:2786(lp_do_section)

If you want I can attach log.smbd file here!

Reply:
Thanks for the explanation.